SampleCode/Python/sample_figure_understanding.ipynb

```diff
@@ -0,0 +1,563 @@
+    "    \"\"\"\n",
+    "    Generates a description for an image using the GPTv model.\n",
+    "\n",
+    "    Parameters:\n",
+    "    - api_base (str): The base URL of the API.\n",
+    "    - api_key (str): The API key for authentication.\n",
+    "    - deployment_name (str): The name of the deployment.\n",
+    "    - api_version (str): The version of the API.\n",
+    "    - image_path (str): The path to the image file.\n",
+    "    - caption (str): The caption for the image.\n",
+    "\n",
+    "    Returns:\n",
+    "    - img_description (str): The generated description for the image.\n",
+    "    \"\"\"\n",
+    "    client = AzureOpenAI(\n",
+    "        api_key=api_key,  \n",
+    "        api_version=api_version,\n",
+    "        base_url=f\"{api_base}/openai/deployments/{deployment_name}\"\n",
+    "    )\n",
+    "\n",
+    "    data_url = local_image_to_data_url(image_path)\n",
+    "\n",
+    "    # We send both image caption and the image body to GPTv for better understanding\n",
+    "    if caption != \"\":\n",
+    "        response = client.chat.completions.create(\n",
+    "                model=deployment_name,\n",
+    "                messages=[\n",
+    "                    { \"role\": \"system\", \"content\": \"You are a helpful assistant.\" },\n",
+    "                    { \"role\": \"user\", \"content\": [  \n",
+    "                        { \n",
+    "                            \"type\": \"text\", \n",
+    "                            \"text\": f\"Describe this image (note: it has image caption: {caption}):\" \n",
+    "                        },\n",
+    "                        { \n",
+    "                            \"type\": \"image_url\",\n",
+    "                            \"image_url\": {\n",
+    "                                \"url\": data_url\n",
+    "                            }\n",
+    "                        }\n",
+    "                    ] } \n",
+    "                ],\n",
+    "                max_tokens=2000 \n",
+    "            )\n",
+    "\n",
+    "    else:\n",
+    "        response = client.chat.completions.create(\n",
+    "            model=deployment_name,\n",
+    "            messages=[\n",
+    "                { \"role\": \"system\", \"content\": \"You are a helpful assistant.\" },\n",
+    "                { \"role\": \"user\", \"content\": [  \n",
+    "                    { \n",
+    "                        \"type\": \"text\", \n",
+    "                        \"text\": \"Describe this image:\" \n",
+    "                    },\n",
+    "                    { \n",
+    "                        \"type\": \"image_url\",\n",
+    "                        \"image_url\": {\n",
+    "                            \"url\": data_url\n",
+    "                        }\n",
+    "                    }\n",
+    "                ] } \n",
+    "            ],\n",
+    "            max_tokens=2000 \n",
```

Comment:
should we define the max_tokens as a CONST?

Reply:
added